src/Proto.Cluster/Member/MemberList.cs

```diff
@@ -68,16 +70,18 @@ public MemberList(Cluster cluster)
 
         public ImmutableHashSet<string> GetMembers() => _activeMembers.Members.Select(m=>m.Id).ToImmutableHashSet();
 
-        public async Task TopologyConsensus()
+        public async Task<bool> TopologyConsensus(CancellationToken ct)
         {
-            while (true)
+            while (!ct.IsCancellationRequested)
```

Comment:
Add silent CT timeout for TopologyConsensus task. prevent this from hanging indefinitely if cluster goes bad